source/collections/VectorList.ooc

```diff
@@ -121,9 +121,14 @@ VectorList: class <T> {
 	}
 	getSlice: func (range: Range) -> This<T> {
 		result := This<T> new(range count)
-		result _count = range count
-		source := (this pointer as Long + (range min * (T size))) as Pointer
-		memcpy(result pointer, source, range count * (T size))
+		this getSliceInto(range, result&)
 		result
 	}
+	getSliceInto: func (range: Range, buffer: This<T>@) {
+		if (buffer _vector capacity < range count)
+			buffer _vector resize(range count)
+		buffer _count = range count
+		source := (this pointer as Long + (range min * (T size))) as Pointer
```

Comment:
Should work anyway, 32 bit *pointer* value will be cast to bigger 64 bit datatype - we do not lose anything here. As long as 'range min * T size' won't exceed 2^32 we should be ok with casting this value back to 32 bit pointer type.
Btw. code for 32-bit machine can be forced with '-m32' flag passed to gcc, so we can check that.